lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -592,13 +595,33 @@ spec = do
                 , "1 +1x + 1y"
                 , "1 + 1x+ 1y"
                 , "1 + 1x +1y"
+                , "xxxx"
+                , "1 + 6667y + 12x"
                 ]
         forM_ invalidFeePolicyTexts $ \policyText ->
             it ("fail fromText @FeePolicy " <> show policyText) $ do
                 let err =
                         "Unable to decode FeePolicy: \
                         \Linear equation not in expected format: a + bx + cy"
                 fromText @FeePolicy policyText === Left (TextDecodingError err)
+        let policyFeesNoFloatingNb =
+                [ "a + bx + cy"
+                , "dasd + asdax + dadsy" ]
+
+        forM_ policyFeesNoFloatingNb $ \policyText ->
+            it ("fail fromText @FeePolicy " <> show policyText) $ do
+                let err = "Expecting floating number"
+                fromText @FeePolicy policyText === Left (TextDecodingError err)
```

Comment:
This error doesn't look right to me ... What exactly is expected to be a "floating number" :thinking: ? (that's a rhetorical question, users shouldn't be presented with this kind of error IMO).

Reply:
@piotr-iohk any thoughts on this? To me it is an "error" in the fee policy `fromText` parser which doesn't properly catch this kind of error.

Reply:
Yes, I agree. I wanted to try fixing this myself...

Reply:
I fixed it. I think...